cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -327,6 +334,11 @@ public ApplicationWithPrograms updateApp(ApplicationId appId, AppRequest appRequ
       newArtifactId = new ArtifactId(currentArtifact.getName(), requestedVersion, currentArtifact.getScope());
     }
 
+    // ownerAdmin.getOwner will give the owner of application irrespective of the version
+    boolean equals = Objects.equals(ownerAdmin.getOwner(appId), appRequest.getOwnerPrincipal());
```

Comment:
you do check that is the app deployment stage, too. Is there a reason for validating it twice?